tests/test_packages/test_connections/test_p2p_libp2p/test_public_dht.py

```diff
@@ -60,19 +64,121 @@ def setup_class(cls):
 
         cls.log_files = []
 
-    def test_reachable(self):
-        public_nodes_maddrs = [PUBLIC_DHT_P2P_MADDR_1, PUBLIC_DHT_P2P_MADDR_2]
-        for maddr in public_nodes_maddrs:
+    def test_connectivity(self):
+        for maddr in PUBLIC_DHT_MADDRS:
             connection = _make_libp2p_connection(
                 DEFAULT_PORT + 1, relay=False, entry_peers=[maddr]
             )
             multiplexer = Multiplexer([connection])
             self.log_files.append(connection.node.log_file)
             multiplexer.connect()
 
-            assert connection.is_connected is True
+            assert (
```

Comment:
If your tests go wrong (assertions don't pass) the multiplexer will not disconnect. Either have individual test classes and do disconnect in teardown so it always happens or wrap in try finally blocks.

Reply:
Indeed